File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    
    li = deque([])
    for i in range(len(priorities)): li.append((i, priorities[i]))
    
    cnt = 0
    while li:
        tmp = li.popleft()
        for i in range(len(li)):
            if li[i][1] > tmp[1]:
                li.append(tmp)
                break
        else:
            cnt += 1
            if tmp[0] == location:
                return cnt

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <deque>

using namespace std;

int solution(vector<int> priorities, int location) {    
    deque<pair<int ,int>> dq;   // {우선순위, 인덱스}
    
    for (int i = 0; i < priorities.size(); ++i)
    {
        dq.push_back({priorities[i], i});
    }
    
    int order = 1;
    while (!dq.empty())
    {
        pair<int, int> cur_pair = dq.front();
        dq.pop_front();
        
        bool flag = true;
        for (int i = 0; i < dq.size(); ++i)
        {
            if (cur_pair.first < dq[i].first)
            {
                dq.push_back(cur_pair);
                flag = false;
                break;
            }
        }
        if (flag)
        {
            if (cur_pair.second == location)
            {
                return order;
            }
            order++;
        }
    }
    
}